### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by niall-byrne** [niall-byrne/goog-playcounts2#5]
I should increase caching times, as updates are happening on a daily cron basis, and we can serve cached data more frequently.

Kubernetes deployment can be tweaked to allocate more CPU where required for performance.

Django needs to optimize the way it's serving its static content, this should also help.

We are templating alot of stuff that could be done with ajax, it may be more performant to send data to the client and have them render the content.

**Comment by niall-byrne** [comment]
I've also made extensive use of pagination on some of the longer pages to improve load times, has made a huge difference.

**Comment by niall-byrne** [comment]
https://codepen.io/dongguangming/pen/amQqky

**Comment by niall-byrne** [comment]
The ajax calls, Django caching, and pagination have all improved the performance dramatically.
I made a note of a great pagination solution above. (I opted to roll my own since my javascript is a bit out of practice and it was fun to do, but the library is a much slicker solution.)  I'm closing this issue, with the intention of iterating on this to further tweak performance.

Ideas for further performance enhancements:
- javascript load order can be optimized
- minification
- per page ajax calls to request smaller data -- maybe a single call to load all data is actually not bad
- host third party libraries and serve them statically
- nginx caching on a proper static folder
